src/openms/include/OpenMS/METADATA/ID/IdentificationData.h

```diff
@@ -0,0 +1,556 @@
+    using DataQueries = IdentificationDataInternal::DataQueries;
+    using DataQueryRef = IdentificationDataInternal::DataQueryRef;
+
+    using ParentMolecule = IdentificationDataInternal::ParentMolecule;
+    using ParentMolecules = IdentificationDataInternal::ParentMolecules;
+    using ParentMoleculeRef = IdentificationDataInternal::ParentMoleculeRef;
+
+    using MoleculeParentMatch = IdentificationDataInternal::MoleculeParentMatch;
+    using ParentMatches = IdentificationDataInternal::ParentMatches;
+
+    using IdentifiedPeptide = IdentificationDataInternal::IdentifiedPeptide;
+    using IdentifiedPeptides = IdentificationDataInternal::IdentifiedPeptides;
+    using IdentifiedPeptideRef =
+      IdentificationDataInternal::IdentifiedPeptideRef;
+
+    using IdentifiedCompound = IdentificationDataInternal::IdentifiedCompound;
+    using IdentifiedCompounds = IdentificationDataInternal::IdentifiedCompounds;
+    using IdentifiedCompoundRef =
+      IdentificationDataInternal::IdentifiedCompoundRef;
+
+    using IdentifiedOligo = IdentificationDataInternal::IdentifiedOligo;
+    using IdentifiedOligos = IdentificationDataInternal::IdentifiedOligos;
+    using IdentifiedOligoRef = IdentificationDataInternal::IdentifiedOligoRef;
+
+    using PeakAnnotations = IdentificationDataInternal::PeakAnnotations;
+    using IdentifiedMoleculeRef =
+      IdentificationDataInternal::IdentifiedMoleculeRef;
+
+    using MoleculeQueryMatch = IdentificationDataInternal::MoleculeQueryMatch;
+    using MoleculeQueryMatches =
+      IdentificationDataInternal::MoleculeQueryMatches;
+    using QueryMatchRef = IdentificationDataInternal::QueryMatchRef;
+
+    // @TODO: allow multiple sets of groups, like with parent molecules
+    // ("ParentMoleculeGroupings")?
+    using QueryMatchGroup = IdentificationDataInternal::QueryMatchGroup;
+    using QueryMatchGroups = IdentificationDataInternal::QueryMatchGroups;
+    using MatchGroupRef = IdentificationDataInternal::MatchGroupRef;
+
+    using ParentMoleculeGroup = IdentificationDataInternal::ParentMoleculeGroup;
+    using ParentMoleculeGroups =
+      IdentificationDataInternal::ParentMoleculeGroups;
+    using ParentGroupRef = IdentificationDataInternal::ParentGroupRef;
+    using ParentMoleculeGrouping =
+      IdentificationDataInternal::ParentMoleculeGrouping;
+    using ParentMoleculeGroupings =
+      IdentificationDataInternal::ParentMoleculeGroupings;
+
+    using AddressLookup = boost::unordered_set<uintptr_t>;
+
+
+    /// Default constructor
+    IdentificationData():
+      current_step_ref_(processing_steps_.end())
+    {
+    }
+
+    // Copy constructor - not allowed, as references would be invalidated:
+    // @TODO: implement using deep copy
+    IdentificationData(const IdentificationData& other) = delete;
+
+    /// Move constructor
+    IdentificationData(IdentificationData&& other):
+      current_step_ref_(other.current_step_ref_)
```

Comment:
can you replace this with  
```
IdentificationData(IdentificationData&& other) noexcept = default;
```

Reply:
PS: it should probably be

```
       current_step_ref_(std::move(other.current_step_ref_) )
```

Reply:
Good point. I've changed it to `... = default;` and it works.